File: java/omnia/data/structure/mutable/MutableSet.java
package omnia.data.structure.mutable;

import omnia.data.structure.Set;

public interface MutableSet<E> extends Set<E>, MutableCollection<E> {

  static <E> MutableSet<E> masking(java.util.Set<E> javaSet) {
    return new MaskingSet<>(javaSet);
  }
}

File: src/omnia/data/structure/mutable/MutableSet.kt
package omnia.data.structure.mutable

import omnia.data.structure.Set

interface MutableSet<E : Any> : Set<E>, MutableCollection<E> {
  companion object {

    fun <E : Any> masking(kotlinSet: kotlin.collections.MutableSet<E>): MutableSet<E> {
      return MaskingSet(kotlinSet)
    }
  }
}